plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1124,6 +1124,17 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
          //
          // For the time being, when `deep-mind = true` is activated, we set `stdout` here to
          // be an unbuffered I/O stream.
+
+         //verify configuration is correct
+         EOS_ASSERT( my->chain_config->read_mode == db_read_mode::HEAD, plugin_config_exception, 
+            "read-mode must be set to head in order to enable deep-mind logging.");
```

Comment:
This disallows running deep-mind in irreversible mode. Unless we know it doesn't work in irreversible mode, this should be removed. In `main` the only two modes are `HEAD` and `IRREVERSIBLE`.

Reply:
Dfuse docs suggests using read-mode = HEAD, which is what we have been using in all of our production setups, but we haven't tried running it with IRREVERSIBLE.

<img width="783" alt="194128707-d6c1173e-738e-488a-8af1-48d8d4d382c6" src="https://user-images.githubusercontent.com/6424172/203105464-b39dfad5-38a2-48a6-946a-ffd22477974c.png">

Will remove the read_mode check, we can revisit this if someone runs into trouble running it in irreversible mode.